src/order/complete_lattice.lean

```diff
@@ -851,6 +851,10 @@ lemma binfi_inf {p : ι → Prop} {f : Π i (hi : p i), α} {a : α} (h : ∃ i,
 by haveI : nonempty {i // p i} := (let ⟨i, hi⟩ := h in ⟨⟨i, hi⟩⟩);
   rw [infi_subtype', infi_subtype', infi_inf]
 
+lemma bsupr_sup {ι : Sort*} {p : ι → Prop} {f : Π i, p i → α} {a : α} (hp : ∃ i, p i) :
```

Comment:
Could you place this above binfi_inf instead of below, to respect the "sup then inf" order of other lemmas in this file?
Also while you are adding lemmas here, could you add sup_bsupr as well?